src/middlewares/autosaveMiddleware.ts

```diff
@@ -21,6 +21,35 @@ const saveFormMiddleware = ({ getState, dispatch }: MiddlewareAPI<Dispatch<AnyAc
             const needToSaveTableChanges = isSendOperation && isAnotherBc
             const selectedCell = state.view.selectedCell
             const isSelectTableCellInit = action.type === types.selectTableCellInit
+
+
+            /**
+             * Default save operation as custom action
+             *
+             * If widget have only custom actions, `defaultSave` option mean witch action
+             * must be executed as save record.
+             * Work only user initial `changeLocation` and don't work at postAction `drillDown`
+             * @param ignorePostAction Indicate that postAction not needed (postAction.drillDown as example)
+             */
+            const defaultSaveWidget = state.view.widgets.find(item => item?.options?.actionGroups?.defaultSave)
+            const defaultCursor = state.screen.bo.bc?.[defaultSaveWidget?.bcName]?.cursor
+            const pendingData = state.view?.pendingDataChanges?.[defaultSaveWidget?.bcName]?.[defaultCursor]
+            if (defaultSaveWidget && action.type === types.changeLocation && !action.ignoreAutosave && pendingData) {
+                action.ignoreAutosave = true
```

Comment:
I think `ignoreAutosave` essentially means "all save operations should not fire autosave middleware".  
Try `actionRole: 'save'` for custom autosave actions and `matchOperationRole` to check for early exit from middleware:
https://github.com/tesler-platform/tesler-ui/blob/1c94589c7d9c1ff06a5220f775302e44f97a9956/src/utils/operations.ts#L43-L52

Reply:
resolved with `defaultSave` widget meta option and without `ignoreAutosave`, `ignorePostAction` flags.